test/controllers/accountingperiod/AccountingPeriodCheckYourAnswersControllerSpec.scala

```diff
@@ -0,0 +1,142 @@
+
+    act.like(journeyRecoveryPage(onPageLoad).updateName("onPageLoad " + _))
+    act.like(journeyRecoveryPage(onSubmit).updateName("onSubmit" + _))
+
+    "viewModel" - {
+
+      val viewModel = AccountingPeriodCheckYourAnswersController.viewModel _
+
+      "have the correct message key for title" in {
+
+        forAll(srnGen, dateRangeWithinRangeGen(dateRange)) { (srn, dateRange) =>
+          viewModel(srn, refineMV[OneToThree](1), dateRange, NormalMode).title.key mustBe "checkYourAnswers.title"
+        }
+      }
+
+      "have the correct message key for heading" in {
+
+        forAll(srnGen, dateRangeWithinRangeGen(dateRange)) { (srn, dateRange) =>
+          messageKey(viewModel(srn, refineMV[OneToThree](1), dateRange, NormalMode).heading) mustBe "checkYourAnswers.heading"
+        }
+      }
+
+      "have the correct message key for start date" in {
+
+        forAll(srnGen, dateRangeWithinRangeGen(dateRange)) { (srn, dateRange) =>
+          viewModel(srn, refineMV[OneToThree](1), dateRange, NormalMode).page.sections
+            .flatMap(_.rows.map(_.key.key)) must contain(
+            "site.startDate"
+          )
+        }
+      }
+
+      "have the correct message key for end date" in {
+
+        forAll(srnGen, dateRangeWithinRangeGen(dateRange)) { (srn, dateRange) =>
+          viewModel(srn, refineMV[OneToThree](1), dateRange, NormalMode).page.sections
+            .flatMap(_.rows.map(_.key.key)) must contain(
+            "site.endDate"
+          )
+        }
+      }
+
+      "have the correct start date" in {
+
+        forAll(srnGen, dateRangeWithinRangeGen(dateRange)) { (srn, dateRange) =>
+          viewModel(srn, refineMV[OneToThree](1), dateRange, NormalMode).page.sections
+            .flatMap(_.rows.collect(_.value match {
+              case m: Message => m.key
+            })) must contain(
+            dateRange.from.show
+          )
+        }
+      }
+
+      "have the correct end date" in {
+
+        forAll(srnGen, dateRangeWithinRangeGen(dateRange)) { (srn, dateRange) =>
+          viewModel(srn, refineMV[OneToThree](1), dateRange, NormalMode).page.sections
+            .flatMap(_.rows.collect(_.value match {
+              case m: Message => m.key
+            })) must contain(dateRange.to.show)
+        }
+      }
+
+//      "have the correct actions" in {
+//
+//        forAll(srnGen, dateRangeWithinRangeGen(dateRange)) { (srn, dateRange) =>
+//          val content = Message("site.change")
+//          val href = routes.AccountingPeriodController.onPageLoad(srn, refineMV[OneToThree](1), NormalMode).url
+//
+//          val actions = List(
+//            SummaryAction(content, href, Message("site.startDate")),
+//            SummaryAction(content, href, Message("site.endDate"))
+//          )
+//
+//          viewModel(srn, refineMV[OneToThree](1), dateRange, NormalMode).page.sections
+//            .flatMap(_.rows.flatMap(_.actions)) must contain allElementsOf actions
+//        }
+//      }
```

Comment:
remove?

Reply:
Ah good catch, these tests should be ok now let me try it quickly :)